agent/token_store.go

```diff
@@ -0,0 +1,84 @@
+package agent
+
+import (
+	"sync"
+)
+
+// TokenStore is used to hold the special ACL tokens used by Consul agents. It
+// is designed to update the tokens on the fly, so the token store itself should
+// be plumbed around and used to get tokens at runtime, don't save the resulting
+// tokens.
+type TokenStore struct {
```

Comment:
As long as the agent code is not reaching into the private vars we're good. A separate package guarantees that, though.